google-cloud-storage/src/main/java/com/google/cloud/storage/spi/v1/HttpStorageRpc.java

```diff
@@ -832,6 +833,10 @@ public String open(StorageObject object, Map<Option, ?> options) {
     Span span = startSpan(HttpStorageRpcSpans.SPAN_NAME_OPEN);
     Scope scope = tracer.withSpan(span);
     try {
+      String kmsKey = object.getKmsKeyName();
+      if (kmsKey != null && kmsKey.contains(CRYPTO_KEY_VERSIONS)) {
+        object.setKmsKeyName(kmsKey.substring(0, kmsKey.indexOf(CRYPTO_KEY_VERSIONS)));
```

Comment:
Oh, instead unset the kms key. Additionally document this in BlobInfo setKmsKeyName java doc.

Reply:
done

Reply:
Whoops, I meant that you should update the metadata at this point to remove KmsKeyName if CryptoKeyVersion is supplied  *and* add documentation on this quirk.

Reply:
done